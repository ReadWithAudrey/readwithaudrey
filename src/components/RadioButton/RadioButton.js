import React from 'react'
import PropTypes from 'prop-types'
import { StyledRadioButton, Label } from './RadioButton.style'

const RadioButton = props => {
  return (
    <React.Fragment>
      <StyledRadioButton
        type="radio"
        value={props.children}
        name={props.children}
      />
      <Label>{props.children}</Label>
    </React.Fragment>
  )
}

RadioButton.propTypes = {
  children: PropTypes.string,
}

export default RadioButton
