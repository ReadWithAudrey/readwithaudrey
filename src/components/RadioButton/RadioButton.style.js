import styled from 'styled-components'

export const StyledRadioButton = styled.section.attrs({
  className: 'montserrat fw4 f6 pv1',
})`
  color: #fff;
`

export const RadioInput = styled.input`
  color: #fff;
`

export const Label = styled.span`
  color: #000;
  padding: 0 1%;
`
