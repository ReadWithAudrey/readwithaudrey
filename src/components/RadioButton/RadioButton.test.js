import React from 'react'
import renderer from 'react-test-renderer'
import RadioButton from './RadioButton'

describe('RadioButton component snapshot test', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<RadioButton />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
