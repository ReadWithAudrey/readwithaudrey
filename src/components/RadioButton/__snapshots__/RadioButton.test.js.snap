// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component snapshot test renders correctly 1`] = `
<section
  className="sc-bdVaJa montserrat fw4 f5 pv1 kXCcmn"
>
  <input
    className="sc-bwzfXH gdcgOT"
    type="radio"
  />
  <span
    className="sc-htpNat hqmdqV"
  />
</section>
`;
